pkg/skaffold/debug/debug.go

```diff
@@ -17,145 +17,21 @@ limitations under the License.
-			}
-			if logrus.IsLevelEnabled(logrus.DebugLevel) {
-				logrus.Debugln("Applied debugging transform:\n", string(manifest))
-			}
-		}
-		updated = append(updated, manifest)
-	}
-
-	return updated, nil
-}
-
-// findArtifact finds the corresponding artifact for the given image.
-// If `builds` is empty, then treat all `image` images as a build artifact.
-func findArtifact(image string, builds []graph.Artifact) *graph.Artifact {
-	if len(builds) == 0 {
-		logrus.Debugf("No build artifacts specified: using image as-is %q", image)
-		return &graph.Artifact{ImageName: image, Tag: image}
-	}
-	for _, artifact := range builds {
-		if image == artifact.ImageName || image == artifact.Tag {
-			logrus.Debugf("Found artifact for image %q", image)
-			return &artifact
-		}
-	}
-	return nil
+type Debugger interface {
+	Start(context.Context, []string) error
+	Stop()
 }
 
-// retrieveImageConfiguration retrieves the image container configuration for
-// the given build artifact
-func retrieveImageConfiguration(ctx context.Context, artifact *graph.Artifact, insecureRegistries map[string]bool) (imageConfiguration, error) {
-	// TODO: use the proper RunContext
-	apiClient, err := docker.NewAPIClient(&runcontext.RunContext{
-		InsecureRegistries: insecureRegistries,
-	})
-	if err != nil {
-		return imageConfiguration{}, fmt.Errorf("could not connect to local docker daemon: %w", err)
-	}
-
-	// the apiClient will go to the remote registry if local docker daemon is not available
-	manifest, err := apiClient.ConfigFile(ctx, artifact.Tag)
-	if err != nil {
-		logrus.Debugf("Error retrieving image manifest for %v: %v", artifact.Tag, err)
-		return imageConfiguration{}, fmt.Errorf("retrieving image config for %q: %w", artifact.Tag, err)
-	}
+type NoopDebugger struct{}
 
-	config := manifest.Config
-	logrus.Debugf("Retrieved local image configuration for %v: %v", artifact.Tag, config)
-	// need to duplicate slices as apiClient caches requests
-	return imageConfiguration{
-		artifact:   artifact.ImageName,
-		env:        envAsMap(config.Env),
-		entrypoint: dupArray(config.Entrypoint),
-		arguments:  dupArray(config.Cmd),
-		labels:     dupMap(config.Labels),
-		workingDir: config.WorkingDir,
-	}, nil
-}
-
-// envAsMap turns an array of environment "NAME=value" strings into a map
-func envAsMap(env []string) map[string]string {
-	result := make(map[string]string)
-	for _, pair := range env {
-		s := strings.SplitN(pair, "=", 2)
-		result[s[0]] = s[1]
-	}
-	return result
-}
-
-func dupArray(s []string) []string {
-	if len(s) == 0 {
-		return nil
-	}
-	dup := make([]string, len(s))
-	copy(dup, s)
-	return dup
-}
-
-func dupMap(s map[string]string) map[string]string {
-	if len(s) == 0 {
-		return nil
-	}
-	dup := make(map[string]string, len(s))
-	for k, v := range s {
-		dup[k] = v
-	}
-	return dup
-}
+func (n *NoopDebugger) Start(_ context.Context, _ []string) error { return nil }
+func (n *NoopDebugger) Stop()                                     {}
```

Comment:
nit: add a newline between the functions, the linter won't add all these whitespaces. 😅